src/java.base/share/classes/java/util/zip/ZipInputStream.java

```diff
@@ -38,9 +38,34 @@
 import static java.util.zip.ZipUtils.*;
 
 /**
- * This class implements an input stream filter for reading files in the
- * ZIP file format. Includes support for both compressed and uncompressed
- * entries.
+ * An input stream for reading compressed and uncompressed
+ * {@linkplain ZipEntry ZIP file entries} from a stream of bytes in the ZIP file
+ * format.
+ *
+ * <H2>Reading Zip File Entries</H2>
+ *
+ * The {@link #getNextEntry()} method is used to read the next ZIP file entry
+ * (Local file (LOC) header record in the ZIP format) and position the stream at
+ * the entry's file data. The file data may read using one of the
+ * {@code ZipInputStream} read methods such
+ * as {@link #read(byte[], int, int) read} or {@link #readAllBytes() readAllBytes()}.
+ * For example:
+ *    {@snippet :
+ *      try (FileInputStream fis = new FileInputStream(jar.toFile());
+ *           ZipInputStream zis = new ZipInputStream(fis)) {
+ *          ZipEntry ze;
+ *          while((ze= zis.getNextEntry()) != null) {
+ *             var bytes = zis.readAllBytes();
+ *             System.out.printf("Entry: %s, bytes read: %s%n", ze.getName(),
+ *                     bytes.length);
+ *          }
+ *      }
+ *    }
+ * The LOC header contains metadata about the Zip file entry. {@code ZipInputStream}
+ * does not read the Central directory (CEN) header for the entry and therefore
+ * will not have access to its metadata such as the external file attributes.
+ * {@linkplain ZipFile} may be used when the information stored within
+ * the CEN header is required.
```

Comment:
I wonder if this paragraph should be an apiNote. It's not wrong as is but it feels like a note to remind developers that this is an "input stream" for iterating through the ZIP entries. The external files attributes are in the CEN at the end of the ZIP file so this is why they aren't available.

Reply:
> I wonder if this paragraph should be an apiNote. It's not wrong as is but it feels like a note to remind developers that this is an "input stream" for iterating through the ZIP entries. The external files attributes are in the CEN at the end of the ZIP file so this is why they aren't available.

converted to an apiNote